src/windows/views/zoom_slider.py

```diff
@@ -0,0 +1,508 @@
+                clip_y = layers.get(clip.data.get('layer', 0), 0) * vertical_factor
+                clip_width = ((clip.data.get('end', 0.0) - clip.data.get('start',0.0))
+                              * pixels_per_second) * horizontal_factor
+                clip_rect = QRectF(clip_x, clip_y, clip_width, 1.0 * vertical_factor)
+                if clip.id in get_app().window.selected_transitions:
+                    # selected clip
+                    self.clip_rects_selected.append(clip_rect)
+                else:
+                    # un-selected clip
+                    self.clip_rects.append(clip_rect)
+
+            for marker in Marker.filter():
+                # Calculate clip geometry (and cache it)
+                marker_x = (marker.data.get('position', 0.0) * pixels_per_second) * horizontal_factor
+                marker_rect = QRectF(marker_x, 0, 0.5, len(layers) * vertical_factor)
+                self.marker_rects.append(marker_rect)
+
+        # Force re-paint
+        self.update()
+
+    def paintEvent(self, event, *args):
+        """ Custom paint event """
+        event.accept()
+
+        # Paint timeline preview on QWidget
+        painter = QPainter(self)
+        painter.setRenderHints(QPainter.Antialiasing | QPainter.SmoothPixmapTransform | QPainter.TextAntialiasing, True)
+
+        # Fill the whole widget with the solid color (background solid color)
+        painter.fillRect(event.rect(), QColor("#191919"))
+
+        # Create pens / colors
+        clip_pen = QPen(QBrush(QColor("#53a0ed")), 1.5)
+        clip_pen.setCosmetic(True)
+        painter.setPen(clip_pen)
+
+        clip_pen = QPen(QBrush(QColor("#53a0ed")), 1.5)
+        clip_pen.setCosmetic(True)
+        selected_clip_pen = QPen(QBrush(QColor("Red")), 1.5)
+        selected_clip_pen.setCosmetic(True)
+
+        scroll_color = QColor("#53a0ed")
+        scroll_color.setAlphaF(0.25)
+        scroll_pen = QPen(QBrush(scroll_color), 2.0)
+        scroll_pen.setCosmetic(True)
+
+        marker_color = QColor("#53a0ed")
+        marker_color.setAlphaF(0.25)
+        marker_pen = QPen(QBrush(marker_color), 1.0)
+        marker_pen.setCosmetic(True)
+
+        playhead_color = QColor("Red")
+        playhead_color.setAlphaF(0.5)
+        playhead_pen = QPen(QBrush(playhead_color), 1.0)
+        playhead_pen.setCosmetic(True)
+
+        handle_color = QColor("#53a0ed")
+        handle_color.setAlphaF(0.65)
+        handle_pen = QPen(QBrush(handle_color), 1.5)
+        handle_pen.setCosmetic(True)
+
+        # Get layer lookup
+        layers = Track.filter()
+
+        # Wait for timeline object and valid scrollbar positions
+        if get_app().window.timeline and self.scrollbar_position[2] != 0.0:
+            # Get max width of timeline
+            project_duration = get_app().project.get("duration")
+            pixels_per_second = event.rect().width() / project_duration
+            project_pixel_width = max(0, project_duration * pixels_per_second)
```

Comment:
You're dividing `event.rect.width()` by `project_duration`, then multiplying that value by `project_duration`? I hate to break it to you, but that's always going to come out as `event.rect().width()`. :laughing:

Reply:
:thinking: